rs/nns/test_utils/golden_nns_state/src/lib.rs

```diff
@@ -43,16 +45,17 @@ pub fn new_state_machine_with_golden_nns_state_or_panic() -> StateMachine {
         PrincipalId::from_str("tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe")
             .unwrap(),
     );
+    // using the canister ranges of both the NNS and II subnets. Note. The
+    // last canister ID in the canister range of the II subnet is omitted so
+    // that the canister range of the II subnet is not used for automatic
+    // generation of new canister IDs.
+    let routing_table = create_routing_table(
+        vec![NNS_CANISTER_ID_RANGE, 0x2100000..=0x21FFFFE],
+        nns_subnet_id,
```

Comment:
The resulting routing table for the NNS test is the same as before. Previously, the flow was as follows:

- Create an empty routing table
- Add the subnet ID (in this case the NNS subnet ID), which [also added the canister range from 0 to 1M](https://sourcegraph.com/github.com/dfinity/ic@714c85c6a4245fb5b39e76f5c8003e6d90e49c4d/-/blob/rs/registry/routing_table/src/lib.rs?L241-257)
- Add the extra canister range, which [added the specified extra canister range to the previously added subnet](https://sourcegraph.com/github.com/dfinity/ic@714c85c6a4245fb5b39e76f5c8003e6d90e49c4d/-/blob/rs/state_machine_tests/src/lib.rs?L1171-1186)

To facilitate the calculation of the canister IDs which need to be added to the non-existent subnet in the routing table, the API was changed to here just provide the canister IDs to the new `create_routing_table` function that generates a routing table which covers all canister IDs (from 0 to `u64::MAX`), where canisters not explicitly set to be part of the subnet we are testing, are set to be part of a new, non-existent subnet.